pl_bolts/datamodules/sklearn_datamodule.py

```diff
@@ -1,280 +1,177 @@
+            and represent the proportion of the dataset to include in the validation split. If int, represents the
+            absolute number of validation samples.
+        test_size: Size of test sample. Splits data and target into training and testing samples. If float, should
+            be between 0.0 and 1.0  and represent the proportion of the dataset to include in the testing split.
+                If int, represents the absolute number of testing samples. Default is None.
+        random_state: Controls the shuffling applied to the data before applying the split. Default is `None`
+        shuffle: Whether to shuffle the data before splitting. If shuffle=False then stratify must be None.
+        stratify: If not None, data is split in a stratified fashion, using this as the class labels.
+        num_workers: Number of subprocesses to use for data loading. 0 means that the data will be loaded in the
+            main process. Number of CPUs available.
+        batch_size: Batch size to use for each dataloader. Default is 1.
+        pin_memory: If `True`, the data loader will copy Tensors into device/CUDA pinned memory before returning
+            them.
+        drop_last: set to `True` to drop the last incomplete batch, if the dataset size is not divisible by the
+            batch size. If `False` and the size of dataset is not divisible by the batch size, then the last batch
+                will be smaller. Default is `False`.
+        persistent_workers: If `True`, the data loader will not shutdown the worker processes after a dataset has
+            been consumed once. This allows to maintain the workers `Dataset` instances alive. Default is `False`.
+
+    Raises:
+        MisconfigurationException: On initialisation, `test_dataset` and `test_size` cannot both be `None`. If a
+            `test_dataset` is provided, it will be used in to create a `test_dataloader`. If a `test_size` is
+            provided then a test `ArrayDataset` will be created from the split of `x` and `y`.
+        MisconfigurationException: On initialisation, if the sum of `val_size` and `test_size` are greater or equal
+            to 1.
+
+    Warnings:
+        `test_dataset` will be used to create the  `test_dataloader` if arguments are provided to `test_dataset` and
+            `test_size`.
+        Training sample will be smaller or equal to the addition of the validation and testing samples if the sum of
+            `val_size` and `test_size` is greater or equal to 0.5.
     """
 
-    name = "sklearn"
-
     def __init__(
         self,
-        X,
-        y,
-        x_val=None,
-        y_val=None,
-        x_test=None,
-        y_test=None,
-        val_split=0.2,
-        test_split=0.1,
-        num_workers=0,
-        random_state=1234,
-        shuffle=True,
-        batch_size: int = 16,
-        pin_memory=True,
-        drop_last=False,
-        *args,
-        **kwargs,
+        data: Any,
+        target: Any,
```

Comment:
Are we really supporting `Any` here?

Reply:
I also agree 😅 I added this as a placeholder for the time being. Are you happy for me to make custom types? If so, where should they live? Perhaps type to encompass' numpy arrays, torch tensors, and lists?